tests/cbmc/stubs/open.c

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ * this file except in compliance with the License. A copy of the License is
+ * located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <assert.h>
+#include <cbmc_proof/nondet.h>
+#include <cbmc_proof/proof_allocators.h>
+#include <sys/fcntl.h>
+
+int open(const char *path, int, ...) {
+    assert(path != NULL);
+    return 0;
```

Comment:
Can't `open` fail?

Reply:
I use this abstraction inside a loop
```c
    do {
        fd = open(file, O_RDONLY);
        S2N_ERROR_IF(fd < 0 && errno != EINTR, S2N_ERR_OPEN);
    } while (fd < 0);
```
If open nondeterminiscally fails, I'll get an unwinding error, correct?

Reply:
Looks like we should also set errno to a nodet value.

What you can do it to have a static. If the loop fails more than once, then succeed/return `errno!=EINTR` the next time.

Reply:
Implemented as suggested.